src/com/oltpbenchmark/DBWorkload.java

```diff
@@ -866,27 +868,25 @@ private static Results runWorkload(List<BenchmarkModule> benchList, boolean verb
 
         }
         Results r = ThreadBench.runRateLimitedBenchmark(workers, workConfs, intervalMonitor);
-        long end = System.nanoTime();
 
         long numNewOrderTransactions = 0;
         for (Worker<?> w : workers) {
             for (LatencyRecord.Sample sample : w.getLatencyRecords()) {
-                if (sample.tranType == newOrderTxnId) {
+                if (sample.tranType == newOrderTxnId && sample.startNs + 1000L * sample.latencyUs <= end) {
```

Comment:
Can this happen? We can start a new txn even after the provided time has passed?
What is considered as txn start time? We should include keying time in that. For example, suppose that:
1. We start tpcc run at time 0.
2. At time 299, we start txn T1 and wait for keying time.
3. At time 310, we actually start the txn execution.
In this scenario, we should consider T1 because the keying time started in the specified time window (0-299).